### DISCUSSION TITLE
Why can't I find the GitHub Codespaces plugin in JetBrains Pycharm 2024.3.1.1(Professional Edition) and JetBrains Gateway 2024.3.1.1?

### DISCUSSION THREAD

**Discussion by zzyu17** [community/community#147971] [Codespaces] [answered]
Question

Just as the pictures seen, I can't find the GitHub Codespaces plugin when searching in the Plugins-Marketplace both in JetBrains Pycharm 2024.3.1.1(Professional Edition) and JetBrains Gateway 2024.3.1.1. 

(Similar searches relevant to "GitHub" and "Codespaces" can't find it too. )

<img width="351" alt="171e3de1cf90bc8b306fe58ccae04d8" src="https://github.com/user-attachments/assets/7529669d-6515-4c1d-8bb9-b3011fbc15e4" />
<img width="1044" alt="5540de0da032ff9509fd714bb2214fd" src="https://github.com/user-attachments/assets/20c7fce5-302c-427e-b9b2-4dfe8211c38d" />
<img width="323" alt="28148d871934793d05f6d5ece18b100" src="https://github.com/user-attachments/assets/d946ca53-93db-46de-a063-0ac3d59ac2ed" />
<img width="611" alt="8ab4837469f30aaaac134903291e820" src="https://github.com/user-attachments/assets/31fae9cb-1aa5-4a6b-8698-5761f974ba47" />

**Comment by ebob9** [comment, marked as answer]
As far as I can tell - it's related updates of the Github codespaces plugin by Github. See my post here https://github.com/orgs/community/discussions/142326#discussioncomment-11524657.

Jetbrains has been updating their IDE and requires new versions of plugins. The Github codespaces plugin is not updated, so it does not work with the latest Jetbrains updates.

It's a maintenance issue - Jetbrains is rapidly updating and Github is not updating at the same pace.

**Reply by zzyu17** [reply]
OIC, thanks for your kind reply. By the way, I have read the attached post of yours, and wonder if it may work if I downgrade my JetBrains Gateway to a certain version that is compatible to install the GitHub CodeSpaces plugin? But even if it may work, as far as I can see, there seems not any previous version of JetBrains Gateway available in the official website of it. (And previous versions of PyCharm are available.)

**Reply by kostrse** [reply]
You can use JetBrains Toolbox to install previous versions

**Reply by zzyu17** [reply]
Thanks! It is possible to install previous versions of JetBrains Gateway via JetBrains Toolbox, however sadly, it is still impossible to install GitHub CodeSpaces plugin in Gateway with versions as early as 2023.1.3 and 2023.2.4.

**Comment by heathdutton** [comment]
The official GitHub Codespaces plugin was quietly discontinued. As an alternative, I built a JetBrains Toolbox plugin that works with PyCharm and other JetBrains IDEs:

https://github.com/heathdutton/github-codespaces-jetbrains-toolbox

Your codespaces show up directly in Toolbox and you can connect from there. Uses `gh` CLI + SSH under the hood.

Or for a quick manual workaround:
```bash
gh codespace ssh -c <codespace-name>
```

**Comment by riteshroshann** [comment]
The disappearance of the plugin isn’t a discovery issue—it reflects the fact that the official GitHub Codespaces plugin for JetBrains was effectively discontinued and never updated for the 2024.x plugin API changes. JetBrains iterated their platform faster than GitHub maintained compatibility, so newer Gateway and PyCharm builds no longer expose the plugin in Marketplace. Downgrading the IDE via Toolbox can help in theory, but in practice even 2023.x builds are past the last plugin-compatible release, which aligns with what you observed. At this point the viable path is either to rely on gh codespace ssh directly or use the community Toolbox integration linked above, which wraps gh + SSH and restores a reasonably native workflow. Out of curiosity, are you primarily trying to use Codespaces for remote computation, for workspace portability, or for enabling the JetBrains remote front-end?

**Comment by hackkan1694** [comment]
https://vnrom.net/bypass/

**Comment by dnuzi** [comment]
Thats awsome
